File: OpenCV_algorithm/convert_rgb2gray/create_fake_image.py
import numpy as np
import cv2

# height, width = 100, 200
height, width = 5, 10
mat = np.zeros((height, width, 3), dtype=np.uint8)

mat[:, :, 0] = 3
mat[:, :, 1] = 3
mat[:, :, 2] = 100

cv2.imwrite("sample.jpg", mat)

File: OpenCV_algorithm/convert_bgr2gray/create_fake_image.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

int main()
{
    int height = 5;
    int width = 10;

    cv::Mat mat(height, width, CV_8UC3);

    for (int y = 0; y < height; y++)
    {
        for (int x = 0; x < width; x++)
        {
            mat.at<cv::Vec3b>(y, x)[0] = 1;
            mat.at<cv::Vec3b>(y, x)[1] = 2;
            mat.at<cv::Vec3b>(y, x)[2] = 3;
        }
    }
    cv::imwrite("sample.jpg", mat);
    return 0;    
}